src/components/modals/investor/add-investor-form.tsx

```diff
@@ -0,0 +1,180 @@
+import { Button } from "@/components/ui/button";
+import { LinearCombobox } from "@/components/ui/combobox";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { StakeholderRelationshipEnum } from "@/prisma/enums";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { StakeholderTypeEnum } from "@prisma/client";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+interface InvestorFormProps {
+  investor?: ZodInvestorType;
+}
+
+const ZodInvestorSchema = z.object({
+  id: z.string().optional(),
+  name: z.string().min(1, { message: "Name is required" }),
+  email: z.string().email().min(1),
+  institutionName: z.string().optional(),
+  investorType: z.nativeEnum(StakeholderTypeEnum, {
+    errorMap: () => ({ message: "Invalid value for investorType" }),
+  }),
+  currentRelationship: z.nativeEnum(StakeholderRelationshipEnum, {
+    errorMap: () => ({ message: "Invalid value for currentRelationship" }),
+  }),
+});
+
+type ZodInvestorType = z.infer<typeof ZodInvestorSchema>;
+
+export const InvestorForm = ({ investor }: InvestorFormProps) => {
+  const investorType =
+    investor?.investorType || investor?.institutionName
+      ? "INSTITUTION"
+      : "INDIVIDUAL";
+
+  const form = useForm<ZodInvestorType>({
+    defaultValues: {
+      name: investor?.name || "",
+      email: investor?.email || "",
+      institutionName: investor?.institutionName || "",
+      investorType,
+      currentRelationship: "INVESTOR",
+    },
+    resolver: zodResolver(ZodInvestorSchema),
+  });
+
+  const isSubmitting = form.formState.isSubmitting;
+
+  const onSubmit = (values: ZodInvestorType) => {
+    // HANDLE THE VALUES :
+
+    console.log("The form values is : ", values);
+  };
+
+  const investorTypeOpts = [
+    { value: "INDIVIDUAL", label: "Individual" },
+    { value: "INSTITUTION", label: "Institution" },
+  ];
```

Comment:
Lets not hardcode these, we need to avoid hardcoding stuff, instead we can bring them in from `prisma/enums.ts`, please do the same on `single-stake-holder-form.tsx` file as well.